analyses/cell-type-consensus/scripts/05-generate-validation-markers.R

```diff
@@ -1,14 +1,75 @@
 #!/usr/bin/env Rscript
 
-# This script is used to create a table of marker genes to help validate the consensus cell type assignments 
+# This script is used to create two tables of marker genes to help validate the consensus cell type assignments 
 # markers are obtained from CellMarker2.0 for Human 
 # Prior to running this script, the CellMarker file must be downloaded using 00-download-cellmarker-ref.sh
 
-# the output is a the `consensus-validation-groups.tsv` file 
-# that contains a list of the top 10 marker genes for each cell ontology ID
+# the outputs are the `consensus-markers.tsv` and `validation-markerts.tsv` file s
+# that contains a list of the top 10 marker genes for each cell ontology ID, one for consensus cell types and one for broad validation groups
 # top marker genes are based on the frequency that marker gene shows up in the total tissues that cell type was found in 
 # for some cell types the marker gene list is longer than 10 and includes any genes with equivalent percentage to the 10th gene in the list 
 
+# Function for formatting top marker gene table --------------------------------
+# takes a table of marker genes and list of ontology Ids and creates a dataframe with 
+# one row per marker gene per ontology Id 
+# also includes how often that gene is found as a marker gene for all included ids
+
+create_top_markers_table <- function(
+    ontology_ids, #list of ids to include marker genes for
+    ontology_column, # prefix for columns that include annotation and ontology Ids for cell types
+    annotation_column, 
+    celltype_groups_df = consensus_groups_df, # dataframe with ontology column and annotation column
+    marker_df = cell_marker_df # data frame with marker genes from cellmarker 
+) {
+  
+  # select unique cell types from 
+  celltypes_df <- celltype_groups_df |> 
+    dplyr::select({{ontology_column}}, {{annotation_column}}) |> 
+    unique()
+  
+  # filter markers to only those for provided ontology IDs
+  filtered_markers_df <- marker_df |> 
+    dplyr::filter(cellontology_id %in% ontology_ids)
+  
+  # get total number of tissues for each cell type 
+  total_tissues_df <- filtered_markers_df |> 
+    dplyr::select(cellontology_id, tissue_type) |> 
+    unique() |>
+    dplyr::count(cellontology_id, name = "celltype_total_tissues")
+  
+  # create data frame with top 10 marker genes per cell type 
+  # ranking is based on the percentage of tissues the gene shows up in for a given cell type  
+  top_markers_df <- filtered_markers_df |> 
+    # first count how many tissues each gene shows up in for each cell type 
+    dplyr::count(cellontology_id, gene_symbol, sort = TRUE, name = "number_of_tissues") |> 
+    # add total possible tissue types 
+    dplyr::left_join(total_tissues_df) |> 
+    dplyr::mutate(
+      # get the total percent of tissues that have that marker gene in that cell type 
+      percent_tissues = round((number_of_tissues/celltype_total_tissues) * 100, 2)
```

Comment:
I don't actually think this is backwards, but I feel like the variable naming could make it clearer that this is not backwards. (I also realize this is the same as before.)

Reply:
`number_of_tissues` is what I think is unclear. I would expect a variable called `number_of_tissues` to be the denominator based on what you're describing in the comment.

Reply:
Ah okay, I added a clarifying comment in https://github.com/AlexsLemonade/OpenScPCA-analysis/pull/1318/commits/55a459f19bd433f9f6b76355ecc970cda361631f describing what each of the variables actually mean.